### ISSUE TITLE
intelligence not work with infer type in 3.7

### ISSUE THREAD

**Issue by charlzyx** [microsoft/TypeScript#35521]
**TypeScript Version:**  3.7.3

**Search Terms:**
intelligence 

**Code**

```ts
/* 
**VS Code Version** 
Version: 1.40.2
Commit: f359dd69833dd8800b54d458f6d37ab7c78df520
Date: 2019-11-25T14:52:45.129Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.0.0
*/
const use = <T extends (...args: any) => any>(
  fn: T,
  params: Parameters<T>[0],
) => {};

const fn = (data: { x: string, y: number }) => {
  return 'something';
};

const yes: Parameters<typeof fn>['0'] = {
  // this way is ok, when enter 'x' , it will intelligence with fn's params[0]
  x: '',
  y: 0,
};

const ohNo = use(fn, {
  // x <---------- Enter x Here
});
```

**Expected behavior:**
when i enter x, give the intelligence of fn params[0] type
![image](https://user-images.githubusercontent.com/18055018/70207727-8beeaf80-1766-11ea-934a-b257c1eee111.png)

**Actual behavior:**
when i enter x, cannot give the intelligence of fn params[0] type, **work well in 3.6.3**
![image](https://user-images.githubusercontent.com/18055018/70207733-8ee9a000-1766-11ea-8b4e-ea9c011f6fac.png)

**Playground Link:** 

**Related Issues:**

**Comment by charlzyx** [comment]
Suprise! it was fixed! at least **3.8.0-dev.20191219** !!

![image](https://user-images.githubusercontent.com/18055018/71232992-1cb0b800-232f-11ea-9501-77b809dc3777.png)


**Thanks for typescript Team!**

And you can use this version by this  extension [ms-vscode.vscode-typescript-next](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next)

**Comment by GuillaumeJasmin** [comment]
I confirm that `3.8.0-dev.20191219` fix my same issues ! 😀
